test/boost/json_cql_query_test.cc

```diff
@@ -495,6 +496,83 @@ SEASTAR_TEST_CASE(test_insert_json_implicit_casts) {
                 "pk": 10, "v": [15]
             }'
         )").get(), marshal_exception);
+
+        // Casting to varchar column:
+        cquery_nofail(e, R"(
+            INSERT INTO implicit_casts JSON '{
+                "pk": 1, "v2": 15
+            }'
+        )");
+        cquery_nofail(e, R"(
+            INSERT INTO implicit_casts JSON '{
+                "pk": 2, "v2": 4.81
+            }'
+        )");
+        cquery_nofail(e, R"(
+            INSERT INTO implicit_casts JSON '{
+                "pk": 3, "v2": 834658736487.7474
+            }'
+        )");
+        cquery_nofail(e, R"(
+            INSERT INTO implicit_casts JSON '{
+                "pk": 4, "v2": 333834658736487.7474
+            }'
+        )");
+        cquery_nofail(e, R"(
+            INSERT INTO implicit_casts JSON '{
+                "pk": 5, "v2": 3333333834658736487.7474
+            }'
+        )");
+        cquery_nofail(e, R"(
+            INSERT INTO implicit_casts JSON '{
+                "pk": 6, "v2": 0.7474
+            }'
+        )");
+        cquery_nofail(e, R"(
+            INSERT INTO implicit_casts JSON '{
+                "pk": 7, "v2": false
+            }'
+        )");
+        cquery_nofail(e, R"(
+            INSERT INTO implicit_casts JSON '{
+                "pk": 8, "v2": 18446744073709551615
+            }'
+        )"); // 2^64 - 1
+
+        require_rows(e, "SELECT pk, v2 FROM implicit_casts", {
+            {int32_type->decompose(1), utf8_type->decompose("15")},
+
+            // Those are better than the output of Scylla 4.0.0 with JsonCpp:
+            // matching the value in INSERT JSON exactly.
```

Comment:
This just goes to show how futile it is to decide our behavior based on some unintentional behavior in Scylla 4.0.0. What does Cassandra do here? What do we think is the **right** thing to do here? :-(

Reply:
Cassandra fails to cast numbers to string:
```
INSERT INTO t JSON '{"pk": 2, "v": 123.32423}';
InvalidRequest: Error from server: code=2200 [Invalid query] message="Error decoding JSON value for v: Expected a UTF-8 string, but got a Double: 123.32423"
```